hack/managed-kyma-migration/migration-testing/migrate-local-moduletemplate.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+# Since we are using relative paths in this script we need to change directory to the script's to not depend on current working directory during the execution
+script_path=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
+cd "$script_path"
+
+# Fetch Kyma CR name managed by lifecycle-manager
+kyma_cr_name=$(kubectl get kyma -n kyma-system --no-headers -o custom-columns=":metadata.name")
+
+# Check if Istio is already present on Kyma CR
+istio_module_count=$(kubectl get kyma "$kyma_cr_name" -n kyma-system -o json | jq '.spec.modules | if . == null then [] else . end | map(. | select(.name=="istio")) | length')
+
+if [  "$istio_module_count" -gt 0 ]; then
+  echo "Istio module already present on Kyma CR, skipping migration"
+  exit 0
+fi
+
+# Fetch all modules, if modules is not defined, fallback to an empty array and count the number modules that have the name "istio"
+
+kyma_cr_modules=$(kubectl get kyma "$kyma_cr_name" -n kyma-system -o json | jq '.spec.modules')
+if [ "$kyma_cr_modules" == "null" ]; then
+  echo "No modules defined on Kyma CR yet, initializing modules array"
+  kubectl patch kyma "$kyma_cr_name" -n kyma-system --type='json' -p='[{"op": "add", "path": "/spec/modules", "value": [] }]'
+fi
+
+(kubectl get -n kyma-system moduletemplate istio-migration-test-fast 2>/dev/null && (echo "Testing module template already on the cluster" ; exit 1)) || echo "Testing ModuleTemplate not found yet. Proceeding to apply one."
```

Comment:
Why do we need this? I think we added this fail fast, because otherwise 
```shell
echo "Proceeding with migration by adding Istio module to Kyma CR $kyma_cr_name"
kubectl patch kyma "$kyma_cr_name" -n kyma-system --type='json' -p='[{"op": "add", "path": "/spec/modules/-", "value": {"name": "istio", "remoteModuleTemplateRef": "kyma-system/istio-migration-test-fast"} }]'
```
can also lead to multiple modules, right?

Reply:
I don't know what would LM do in this case, but Kubernetes would add 2nd Istio to the KymaCR which is sufficient in my opinion to exit early.